File: microservicio-cliente-persona/src/main/java/com/develop/clientepersona/dto/ClienteDTO.java
package com.develop.clientepersona.dto;

import com.develop.clientepersona.enums.GenderPerson;
import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.io.Serializable;
import java.sql.Timestamp;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class ClienteDTO implements Serializable {
    private Long id;
    @CreationTimestamp
    private Timestamp creationDate;
    @NotBlank(message = "El campo nombre no debe estar en blanco")
    private String fullName;
    @NotNull(message = "El género es obligatorio")
    private GenderPerson genderPerson;
    @Positive(message="El campo edad debe ser un valor positivo")
    @Max(value = 120, message="El campo edad debe estar en en un rango de entre 1 y 120 años")
    private int age;
    @NotBlank(message = "El numero de identificación no debe estar en blanco")
    @Size (min = 10,max = 13, message = "El campo identificacion debe tener entre 10 y 13 caracteres")
    private String identification;
    @NotBlank(message = "El campo dirección no debe estar en blanco")
    private String address;
    @NotBlank(message = "El campo teléfono no debe estar en blanco")
    private String phone;
    @NotBlank(message = "El campo password no debe estar en blanco")
    @Size (min = 8,max = 15, message = "El campo password debe tener entre 8 y 15 caracteres")
    private String password;
    private boolean state;
}

File: client-person/Models/Dto/ClienteDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using client_person.Enum;

namespace client_person.Dto
{
    [JsonUnmappedMemberHandling(JsonUnmappedMemberHandling.Disallow)]
    public class ClienteDto
    {
        public long id { get; set; }
        public DateTime creationDate { get; set; }

        [Required(ErrorMessage = "El nombre es obligatorio.")]
        public string? fullName { get; set; }

        [Required(ErrorMessage = "El género es obligatorio.")]
        [EnumDataType(typeof(GenderPerson), ErrorMessage = "Género inválido.")]
        public GenderPerson genderPerson { get; set; }

        [Required(ErrorMessage = "La edad es obligatoria.")]
        [Range(0, 120, ErrorMessage = "La edad debe estar entre 0 y 120.")]
        public int age { get; set; }

        [Required(ErrorMessage = "La identificación es obligatoria.")]
        [RegularExpression(@"^\d{10}(\d{3})?$", ErrorMessage = "La identificación debe tener 10 o 13 dígitos numéricos.")]
        public string? identification { get; set; }
       
        [Required(ErrorMessage = "La dirección es obligatoria.")]
        public string? address { get; set; }
        
        [Required(ErrorMessage = "El teléfono es obligatorio.")]
        [Phone(ErrorMessage = "Número de teléfono inválido.")]
        public string? phone { get; set; }

        [Required(ErrorMessage = "La contraseña es obligatoria.")]
        [MinLength(6, ErrorMessage = "La contraseña debe tener al menos 6 caracteres.")]
        public string? password { get; set; }

        public bool state { get; set; }
        
    }
}